app/components/profile.jsx

```diff
@@ -19,73 +23,60 @@ export class Profile extends Component {
     this.reset();
   }
 
-  componentWillUpdate (props) {
-    this.reset(props);
+  componentWillUpdate ({ params: { username } }) {
```

Comment:
okay so i found a bug where if youre on a profile page (e.g. someone elses like `/u/hiami`) and then you navigate to another profile page directly (e.g. your own by clicking the nav link), it totally flips out (independent of the `NotFoundComponent` bug that i already mentioned). this bug actually already exists in prod for the tracker page, but theres no way to _directly_ navigate to another tracker page from the tracker page (if you edit the url, it causes a refresh which makes it fine).

ill be telling you the lines to edit (and why) for both the profile page and the tracker page.

you should change it from `componentWillUpdate` to `componentWillReceiveProps`. since this function is calling `this.reset()`, it will also be calling `this.setState` (line 35), but turns out ["you cannot call `this.setState()` here. If you need to update state in response to a prop change, use `componentWillReceiveProps()` instead."](https://facebook.github.io/react/docs/react-component.html#componentwillupdate) so there you go.

also, you should change it from `{ params: { username } }` to `nextProps`. the main reason why is because we need to the object intact cause we'll be using it later. ill explain more when we get there.

final result:

```js
componentWillReceiveProps (nextProps) {
```

you should also do this in `tracker.jsx` (line 28)